packages/tabs/tabs.ts

```diff
@@ -0,0 +1,276 @@
+
+      const left = activeTabRect.left - tabListRect.left;
+      const width = activeTabRect.width;
+
+      this.selectionIndicator.style.left = `${left}px`;
+      this.selectionIndicator.style.width = `${width}px`;
+    });
+  }
+
+  private updatePanels() {
+    // Update tab panels visibility
+    const panels = document.querySelectorAll('w-tab-panel');
+    panels.forEach((panel) => {
+      const panelName = panel.getAttribute('name');
+      if (panelName === this._activeTab) {
+        panel.removeAttribute('hidden');
+        // Also explicitly set hidden property to false for web components
+        (panel as HTMLElement).hidden = false;
+      } else {
+        panel.setAttribute('hidden', '');
+        (panel as HTMLElement).hidden = true;
+      }
+    });
+
+    // Update tab active states
+    const tabs = this.querySelectorAll('w-tab');
+    tabs.forEach((tab) => {
+      const tabName = tab.getAttribute('name');
+      if (tabName === this._activeTab) {
+        tab.setAttribute('active', '');
+      } else {
+        tab.removeAttribute('active');
+      }
+    });
+  }
+
+  private _handleKeyDown = (event: KeyboardEvent) => {
+    if (event.altKey || event.ctrlKey || event.shiftKey || !['ArrowLeft', 'ArrowRight', 'Home', 'End'].includes(event.key)) {
+      return;
+    }
+
+    const tabs = Array.from(this.querySelectorAll('w-tab'));
+    const currentIndex = tabs.findIndex((tab) => tab.getAttribute('name') === this._activeTab);
+
+    if (currentIndex === -1) return;
+
+    let nextIndex = currentIndex;
+
+    switch (event.key) {
+      case 'Home':
+        nextIndex = 0;
+        break;
+      case 'End':
+        nextIndex = tabs.length - 1;
+        break;
+      case 'ArrowLeft':
+        nextIndex = Math.max(0, currentIndex - 1);
+        break;
+      case 'ArrowRight':
+        nextIndex = Math.min(tabs.length - 1, currentIndex + 1);
+        break;
+    }
+
+    if (nextIndex !== currentIndex) {
+      event.preventDefault();
+      const nextTab = tabs[nextIndex];
+      const nextTabName = nextTab.getAttribute('name');
+
+      if (nextTabName) {
+        this.active = nextTabName;
+        this._activeTab = nextTabName;
+        this.updateSelectionIndicator();
+        this.updatePanels();
+        this._notifyTabChange();
+
+        // Focus the next tab
+        (nextTab as HTMLElement).focus();
+      }
+    }
+  };
+
+  private get _tabCount() {
+    return this.querySelectorAll('w-tab').length;
+  }
+
+  private get _gridClass() {
+    const count = Math.min(this._tabCount, 9);
+    return ccGridLayout[`cols${count}` as keyof typeof ccGridLayout] || ccGridLayout.cols1;
+  }
+
+  render() {
+    const navClasses = classNames(this.tabClass, ccTabs.wrapper);
+    const divClasses = classNames([ccTabs.base, this._gridClass]);
+
+    return html`
+      <div class="${navClasses}">
+        <div role="tablist" class="${divClasses}" @keydown="${this._handleKeyDown}">
+          <slot></slot>
```

Comment:
would it make sense to also run update tabs when the slot changes in case w-tabs are added or removed?